app/controllers/alchemy/admin/pages_controller.rb

```diff
@@ -171,9 +171,10 @@ def unlock
 
       def visit
         @page.unlock!
-        redirect_to show_page_path(
+        redirect_to show_page_url(
           urlname: @page.urlname,
-          locale: prefix_locale? ? @page.language_code : nil
+          locale: prefix_locale? ? @page.language_code : nil,
+          host: @page.site.host == "*" ? request.host : @page.site.host
```

Comment:
The other way around:

- Visit your local site via http://localhost:3000
- Change the site via "Site select" to a default local `sitename.dev` site
- Visit the page via "Visit page" button

=> the resulting url is http://sitename.dev:3000/home instead of http://sitename.dev/home

It works, though, the page loads, but, meh.